alert-common/src/main/java/com/synopsys/integration/alert/common/action/api/AbstractConfigResourceActions.java

```diff
@@ -0,0 +1,182 @@
+
+public abstract class AbstractConfigResourceActions implements ResourceActions<FieldModel>, TestAction<FieldModel>, ValidateAction<FieldModel> {
+    private AuthorizationManager authorizationManager;
+    private DescriptorAccessor descriptorAccessor;
+
+    public AbstractConfigResourceActions(AuthorizationManager authorizationManager, DescriptorAccessor descriptorAccessor) {
+        this.authorizationManager = authorizationManager;
+        this.descriptorAccessor = descriptorAccessor;
+    }
+
+    public AbstractConfigResourceActions(AuthorizationManager authorizationManager) {
+        this.authorizationManager = authorizationManager;
+    }
+
+    protected abstract ActionResponse<FieldModel> createResource(FieldModel resource);
+
+    protected abstract ActionResponse<FieldModel> deleteResource(Long id);
+
+    protected abstract ActionResponse<List<FieldModel>> readAllResources();
+
+    protected abstract ActionResponse<List<FieldModel>> readAllByContextAndDescriptor(String context, String descriptorName);
+
+    protected abstract Optional<FieldModel> findFieldModel(Long id);
+
+    protected abstract ValidationActionResponse testResource(FieldModel resource);
+
+    protected abstract ActionResponse<FieldModel> updateResource(Long id, FieldModel resource);
+
+    protected abstract ValidationActionResponse validateResource(FieldModel resource);
+
+    public ActionResponse<List<FieldModel>> getAllByContextAndDescriptor(String context, String descriptorName) {
+        if (!authorizationManager.hasReadPermission(context, descriptorName)) {
+            return new ActionResponse<>(HttpStatus.FORBIDDEN, AbstractResourceActions.FORBIDDEN_MESSAGE);
+        }
+        return readAllByContextAndDescriptor(context, descriptorName);
+    }
+
+    @Override
+    public ActionResponse<FieldModel> create(FieldModel resource) {
+        if (!authorizationManager.hasCreatePermission(resource.getContext(), resource.getDescriptorName())) {
+            return new ActionResponse<>(HttpStatus.FORBIDDEN, AbstractResourceActions.FORBIDDEN_MESSAGE);
+        }
+        ValidationActionResponse validationResponse = validateResource(resource);
+        if (validationResponse.isError()) {
+            return new ActionResponse<>(validationResponse.getHttpStatus(), validationResponse.getMessage().orElse(null));
+        }
+        return createResource(resource);
+    }
+
+    @Override
+    public ActionResponse<List<FieldModel>> getAll() {
+        try {
+            Set<String> descriptorNames = descriptorAccessor.getRegisteredDescriptors()
+                                              .stream()
+                                              .map(RegisteredDescriptorModel::getName)
+                                              .collect(Collectors.toSet());
+            if (!authorizationManager.anyReadPermission(List.of(ConfigContextEnum.DISTRIBUTION.name(), ConfigContextEnum.GLOBAL.name()), descriptorNames)) {
```

Comment:
It might be worthwhile to create an enhancement request for `AuthorizationManager` to have methods that also take collections of enums.

Reply:
Yeah I was going to make that change in this set of code changes but I will add it separately.